h2o-extensions/xgboost/src/main/java/hex/tree/xgboost/XGBoost.java

```diff
@@ -412,7 +417,9 @@ private String createFeatureMapFile(String featureMap) {
       OutputStream os = null;
```

Comment:
I would also modify this method to modify the return String that is then directly assigned to featureMapFileAbsolutePath is unnecesary and unredable, also I would add a check here that would throw illegal state exceptio when this method would be called and featureMapFileAbsolutePath != null or altenatively delete the file when != null before creating new one?
or maybe this needs bit more thought.

Reply:
Thanks, the code needed a clean-up - please see my changes